### PR TITLE
build: activate render3 tests in CI

### PR DESCRIPTION
**Pull request by marclaval**
All render3 tests are passing in all supported browsers, let's keep it that way :)

### CODE REVIEW

File: packages/core/src/render3/instructions.ts [hunk 1]
```diff
@@ -585,9 +593,10 @@ export function elementAttribute(index: number, attrName: string, value: any): v
           (renderer as ProceduralRenderer3).removeAttribute(element.native, attrName) :
           element.native.removeAttribute(attrName);
     } else {
+      const renderValue = value != null ? value.toString() : null;
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: this looks like a call to `stringify` Can we use that instead?

File: packages/core/src/render3/instructions.ts [hunk 2]
```diff
@@ -585,9 +593,10 @@ export function elementAttribute(index: number, attrName: string, value: any): v
...
       (renderer as ProceduralRenderer3).setAttribute ?
-          (renderer as ProceduralRenderer3).setAttribute(element.native, attrName, value) :
-          element.native.setAttribute(attrName, value);
+          (renderer as ProceduralRenderer3).setAttribute(element.native, attrName, renderValue) :
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: Why do we have to stringify? I thought that DOM would do that for us. What happens if we don't?

File: packages/core/test/render3/render_util.ts [hunk 3]
```diff
@@ -35,13 +36,17 @@ requestAnimationFrame.flush = function() {
...
+      document.body.removeChild(containerEl);
+    } catch (e) {
+    }
   }
   containerEl = document.createElement('div');
   containerEl.setAttribute('host', '');
   document.body.appendChild(containerEl);
   host = null;
   // TODO: assert that the global state is clean (e.g. ngData, previousOrParentNode, etc)
+  resetApplicationState();
```
Comment (mhevery) [inline packages/core/test/render3/render_util.ts]: Why do we have to do this? Can't this be a part of the `renderTemplate` method?

File: packages/core/src/render3/instructions.ts [hunk 4]
```diff
@@ -264,6 +264,14 @@ export function createLNode(
 //// Render
 //////////////////////////
 
+/**
+ * Resets the application state.
+ */
+export function resetApplicationState() {
```
Comment (mhevery) [inline packages/core/src/render3/instructions.ts]: can this be part of `renderTemplate` method? It is strange that we need this for the tests.

File: packages/core/src/render3/instructions.ts (hunk 2 above)
Comment (mlaval) [reply packages/core/src/render3/instructions.ts]: They all do stringify using toString(), except IE9 :(

File: packages/core/src/render3/instructions.ts (hunk 1 above)
Comment (mlaval) [reply packages/core/src/render3/instructions.ts]: you're right, I'll change that

File: packages/core/src/render3/instructions.ts (hunk 4 above)
Comment (mlaval) [reply packages/core/src/render3/instructions.ts]: Done

File: packages/core/test/render3/render_util.ts (hunk 3 above)
Comment (mlaval) [reply packages/core/test/render3/render_util.ts]: If we don't, each new host created is appended to the previous one. This creates a huge tree and wrong behaviours which showed up in projection tests.

File: packages/core/test/render3/component_spec.ts [hunk 5]
```diff
@@ -8,10 +8,10 @@
 
 import {ViewEncapsulation} from '../../src/core';
 import {D, E, T, b, defineComponent, e, markDirty, t} from '../../src/render3/index';
-import {createRendererType2} from '../../src/view';
+import {createRendererType2} from '../../src/view/index';
```
Comment (alfaproject) [inline packages/core/test/render3/component_spec.ts]: This looks wrong

File: packages/core/test/render3/renderer_factory_spec.ts [hunk 6]
```diff
@@ -10,11 +10,11 @@ import {AnimationEvent} from '@angular/animations';
 import {MockAnimationDriver, MockAnimationPlayer} from '@angular/animations/browser/testing';
 
 import {RendererType2, ViewEncapsulation} from '../../src/core';
-import {D, E, L, T, b, defineComponent, detectChanges, e, p} from '../../src/render3';
-import {createRendererType2} from '../../src/view';
+import {D, E, L, T, b, defineComponent, detectChanges, e, p} from '../../src/render3/index';
+import {createRendererType2} from '../../src/view/index';
```
Comment (alfaproject) [inline packages/core/test/render3/renderer_factory_spec.ts]: Both of these look wrong

File: packages/core/test/render3/component_spec.ts (hunk 5 above)
Comment (mhevery) [reply packages/core/test/render3/component_spec.ts]: this is required for JSCompiler

File: packages/core/test/render3/renderer_factory_spec.ts (hunk 6 above)
Comment (mhevery) [reply packages/core/test/render3/renderer_factory_spec.ts]: this is required for JSCompiler

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**